liblwgeom/lwstroke.c

```diff
@@ -409,9 +423,16 @@ lwarc_linearize(POINTARRAY *to,
 	if( is_circle )
 	{
 		increment = fabs(increment);
+		segments = ceil(total_angle / increment);
```

Comment:
Is this kind of repeating the work done around line 342?

Reply:
Yes; since it might modify the value of increment I added this again here to make sure there is at least 3 segments for circles.